### PR TITLE
Add initial pass of CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by heeheehee-kolmafia**
There's a lot more room for expansion, but I think this is a good
start.

Draft can be seen [here](https://github.com/kolmafia/kolmafia/blob/c3bb2af48b86a5c92b50a1957f6688c738a1d901/CONTRIBUTING.md).

### CODE REVIEW

File: CONTRIBUTING.md
Comment (jaadams5) [inline CONTRIBUTING.md:26]: I would not complain if this were tweaked.  It was written at a time when the social aspects were prominent - at least one dev (me) was ignoring patches from people who were being rude, annoying and acting as if the fact that they did a little work entitled them to have that work incorporated into the repository.  I think the GitHub workflow with reviews is a good step towards reducing the issues caused by personalities.  I would like to encourage discussions "should we do this?  is this a good idea?" before someone writes code and submits a pull request but we could probably rework this.

File: CONTRIBUTING.md
Comment (jaadams5) [inline CONTRIBUTING.md:44]: Is this still true and what we want or how we expect to do things?  Can non-staff contribute to a review even if they cannot approve the PR?  I'd like the answer to be yes and part of the audition to become staff would include constructive participation in reviews.

File: CONTRIBUTING.md
Comment (heeheehee-kolmafia) [reply CONTRIBUTING.md:9]: Done. I renamed the section, and added a comment regarding "in the spirit of github flow, we do things this way" Let me know if that's any better!

File: CONTRIBUTING.md
Comment (heeheehee-kolmafia) [reply CONTRIBUTING.md]: I've rewritten that initial paragraph to incorporate a lot of the things you mentioned. Thoughts?

File: CONTRIBUTING.md
Comment (heeheehee-kolmafia) [reply CONTRIBUTING.md]: > Can non-staff contribute to a review even if they cannot approve the PR? I'd like the answer to be yes and part of the audition to become staff would include constructive participation in reviews.

Yes, agreed on both fronts. I also added a vague comment about how we may initiate contact, since IIRC that's how I became a dev.

What I don't want is to encourage a system of basically mini-mods where non-devs go around nitpicking others' PRs. For now "constructively engaging" seems good enough, but we can revisit if it becomes an issue.

File: CONTRIBUTING.md
Comment (BadHorseMonkey) [inline CONTRIBUTING.md:29]: Can we link to the “how to write a good bug report?” In this paragraph?

File: CONTRIBUTING.md
Comment (BadHorseMonkey) [inline CONTRIBUTING.md:66]: Some tasks, like jpackage, require Java14.   Should we mention that or just wait until that lands and update this doc?

File: CONTRIBUTING.md
Comment (BadHorseMonkey) [inline CONTRIBUTING.md:44]: Is it still just us two?

File: CONTRIBUTING.md
Comment (BadHorseMonkey) [reply CONTRIBUTING.md:9]: Do we want to link to the (not currently up to date) build and IDE setup guides on the wiki?  Should there be general links to the wiki in the suggested reading section?

**Review by BadHorseMonkey — APPROVED** [review]

**Comment by BadHorseMonkey** [comment]
I approved it, but if you want to tweak it, I’ll re-review.

File: CONTRIBUTING.md
Comment (heeheehee-kolmafia) [reply CONTRIBUTING.md:44]: Depends on the definition. For expanding commit access, I think it's "whoever's listed as an owner in the github organization" which also includes Rinn and Veracity.

File: CONTRIBUTING.md
Comment (heeheehee-kolmafia) [reply CONTRIBUTING.md:66]: Well, it's not necessary to run the jpackage task for general contributors, unless you're working on that feature.

File: CONTRIBUTING.md
Comment (heeheehee-kolmafia) [reply CONTRIBUTING.md:9]: I guess my question on this front is "what do we see as the purpose of the wiki?" I view it primarily as a user guide, not as a developer guide. It could be both, but I think it makes sense to put dev-only content where devs are already looking.

Comment (jaadams5) [reply CONTRIBUTING.md:9]: If someone needs help in setting things up and we want to preserve the interaction so we don't have to repeat ourselves for some visitor from the future, I think there should be some minimal guidance on KoLmafia.us.  We do have at least one person who wants to setup and build locally but has no desire to share what they are doing so that conversation might fit better there then here.  If we can clearly identify something of benefit/interest only to devs then it belongs here.

File: CONTRIBUTING.md
Comment (jaadams5) [reply CONTRIBUTING.md:29]: Do we want to link to that here?  WE aren't reporting bugs on GitHub and reports at KoLmafia are driving activities.

**Comment by jaadams5** [comment]
We can probably tweak this some more but I'm fine with it as it stands.  We can wait until someone else reads it and asks questions :-)

**Review by jaadams5 — APPROVED** [review]
OK by me, at least for round 1.

File: CONTRIBUTING.md
Comment (heeheehee-kolmafia) [reply CONTRIBUTING.md:9]: Hey, if you've already downloaded the repo, you'll have a copy of CONTRIBUTING.md locally :)

**Comment by heeheehee-kolmafia** [comment]
I'll commit this; we can discuss further changes in a followup.

**Pull request merged**
